### PR TITLE
Rails 4.0 support

### PR DESCRIPTION
**Pull request by charliesome**
This pull request adds Rails 3.2 and 4.0 gemfiles, loosens up the gemspec dependency to support ActiveRecord 4.0.

The tests are passing locally. I switched away from the brittle stubs this was relying on before to avoid a persistent database by just using an in-memory SQLite database and testing against that.

cc @technoweenie @github/rails-upgrades

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -16,52 +16,20 @@
...
-  end
-
-  def column_defaults
-    {}
-  end
-
-  def columns_hash
-    {}
-  end
+ActiveRecord::Base.establish_connection(
```
Comment (rsanheim) [inline test/test_helper.rb]: :+1:

File: test/serialized_attributes_test.rb [hunk 2]
```diff
@@ -86,9 +88,9 @@ def setup
     end
 
     test "#attribute_names contains serialized fields" do
-      assert_equal %w(active age average birthday extras lottery_picks names title), @record.attribute_names
+      assert_equal %w(active age average birthday extras id lottery_picks names title), @record.attribute_names
```
Comment (rsanheim) [inline test/serialized_attributes_test.rb]: Why would `id` be in here now?

**Comment by rsanheim** [comment]
We should get this running on travis for some real CI...

File: test/serialized_attributes_test.rb (hunk 2 above)
Comment (charliesome) [inline test/serialized_attributes_test.rb]: This appears to be a recent change in ActiveRecord - the stock `#attribute_names` now includes the `id` column

**Pull request merged**